### PR TITLE
Python binding: rename block types

### LINKED ISSUE #1634: python: rename block types
https://github.com/iotaledger/iota-sdk/issues/1515

### PR DESCRIPTION
**Pull request by Alex6323**
## Links to any relevant issues

Closes #1634

### CODE REVIEW

File: bindings/python/iota_sdk/types/block/block.py [hunk 1]
```diff
@@ -2,15 +2,78 @@
...
+from iota_sdk.types.block.body.basic import BasicBlockBody
+from iota_sdk.types.block.body.validation import ValidationBlockBody
 
 
-class BlockType(IntEnum):
-    """Block types.
+@json
+@dataclass
+class Block:
+    """A signed block that can hold either a `BasicBlockBody` or a `ValidationBlockBody`.
```
Comment (Thoralf-M) [inline bindings/python/iota_sdk/types/block/block.py]:
```suggestion
    """A block that can hold either a `BasicBlockBody` or a `ValidationBlockBody`.
```

**Review by Thoralf-M — DISMISSED** [review]

**Review by Thoralf-M — DISMISSED** [review]

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Remaining instances in
- [x] `08_data_block.py`
- [x] `post_raw_block.py`
- [x] `submit_and_read_block.py`
- [x] `utils.py`
- [x] `test_block.py`

File: bindings/python/iota_sdk/types/block/metadata.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
 from typing import Optional
 from iota_sdk.types.common import HexStr, json
 # TODO rename change to Block
```
Comment (thibault-martinez) [inline bindings/python/iota_sdk/types/block/metadata.py]: remove

File: bindings/python/iota_sdk/types/block/body/kind.py
Comment (thibault-martinez) [inline bindings/python/iota_sdk/types/block/body/kind.py:1]: Don't call the file `kind` if nothing else is called kind

File: bindings/python/iota_sdk/types/block/metadata.py (hunk 2 above)
Comment (Alex6323) [reply bindings/python/iota_sdk/types/block/metadata.py]: Done

File: bindings/python/iota_sdk/types/block/body/kind.py
Comment (Alex6323) [reply bindings/python/iota_sdk/types/block/body/kind.py:1]: renamed to `type.py`

**Review by Thoralf-M — DISMISSED** [review]

File: bindings/python/iota_sdk/types/block/body/kind.py
Comment (thibault-martinez) [reply bindings/python/iota_sdk/types/block/body/kind.py:1]: Is there no import to change then?

**Review by thibault-martinez — APPROVED** [review]

File: bindings/python/iota_sdk/types/block/block.py [hunk 3]
```diff
@@ -2,15 +2,78 @@
...
+    protocol_version: int
+    network_id: int = field(metadata=config(
+        encoder=str
+    ))
+    issuing_time: int = field(metadata=config(
+        encoder=str
+    ))
+    slot_commitment_id: HexStr
+    latest_finalized_slot: SlotIndex
+    issuer_id: HexStr
```
Comment (thibault-martinez) [inline bindings/python/iota_sdk/types/block/block.py:41]: Wait actually this is missing the block header?

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: bindings/python/iota_sdk/types/block/block.py (hunk 3 above)
Comment (Alex6323) [reply bindings/python/iota_sdk/types/block/block.py:41]: yeah, but that should be a quick separate PR as it's not just renaming? will open it right after, or want me to include it?

Comment (thibault-martinez) [reply bindings/python/iota_sdk/types/block/block.py:41]: I thought that would be included here. Fine to do in another PR, let's just not close the issue until it's done though.

**Review by thibault-martinez — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**